compiler/src/dotty/tools/dotc/interactive/InteractiveCompiler.scala

```diff
@@ -14,4 +15,7 @@ class InteractiveCompiler extends Compiler {
   override def phases: List[List[Phase]] = List(
     List(new FrontEnd)
   )
+
+  override def rootContext(implicit ctx: Context) =
+    super.rootContext.fresh.addMode(Mode.Interactive)
```

Comment:
There's also Mode.ReadPositions that we always need in the IDE, it's currently set in InteractiveDriver but could be moved here.